src/backend/opengl/common.cpp

```diff
@@ -170,12 +170,60 @@ Shaders loadShaders(const char* pVertexShaderSrc,
     return out;
 }
 
-GLuint initShaders(const char* pVertShaderSrc, const char* pFragShaderSrc, const char* pGeomShaderSrc)
+namespace opengl
+{
+
+ShaderProgram::ShaderProgram(const char* pVertShaderSrc,
+                             const char* pFragShaderSrc,
+                             const char* pGeomShaderSrc)
+    : mVertex(0), mFragment(0), mGeometry(0), mProgram(0)
 {
     Shaders shrds = loadShaders(pVertShaderSrc, pFragShaderSrc, pGeomShaderSrc);
-    GLuint shaderProgram = glCreateProgram();
-    attachAndLinkProgram(shaderProgram, shrds);
-    return shaderProgram;
+    mProgram = glCreateProgram();
+    attachAndLinkProgram(mProgram, shrds);
+    mVertex = shrds.vertex;
+    mFragment = shrds.fragment;
+    mGeometry = shrds.geometry;
+}
```

Comment:
`mVertex`, `mFragment`, and `mGeometry` are being created / destroyed but are never being used. Do you really need them?